src/lib/CookieDecoder.ts

```diff
@@ -134,20 +144,26 @@ export class CookieDecoder {
         return new Map(accountsMapArray);
     }
 
-    private static decodeAccount(bytes: number[], labelLength?: number): AccountInfoEntry {
-        if (typeof labelLength === 'undefined') labelLength = this.readByte(bytes);
+    private static decodeAccount(bytes: number[], type: WalletType, labelLength?: number): AccountInfoEntry {
+        if (typeof labelLength === 'undefined') {
+            labelLength = this.readByte(bytes);
+        }
 
-        if (labelLength > LABEL_MAX_LENGTH) throw new Error('Malformed Cookie, label too long');
+        if (labelLength > LABEL_MAX_LENGTH) {
+            throw new Error('Malformed Cookie, label too long');
+        }
 
         // Account label
         const labelBytes = this.readBytes(bytes, labelLength);
 
-        let accountLabel: string;
-        if (labelBytes.length === 0) accountLabel = 'Standard Account';
-        else accountLabel = Utf8Tools.utf8ByteArrayToString(new Uint8Array(labelBytes));
+        const accountLabel = labelBytes.length === 0
+            ? type === WalletType.BIP39
+                ? ADDRESS_DEFAULT_LABEL_KEYGUARD
+                : ADDRESS_DEFAULT_LABEL_LEDGER
+            : Utf8Tools.utf8ByteArrayToString(new Uint8Array(labelBytes));
 
         // Account address
-        const addressBytes = this.readBytes(bytes, 20 /* Nimiq.Address.SERIALIZED_SIZE */);
+        const addressBytes = this.readBytes(bytes, Nimiq.Address.SERIALIZED_SIZE);
```

Comment:
Revert this. The iframe does not have the Nimiq lib loaded, that's why I used the plain number.